### PR TITLE
Improve image building scripts

### PR DESCRIPTION
**Pull request by Sunnatillo**
This PR makes following changes:

- bump k8s to v1.30.0
- k8s version controlled in one file
- add prepulling container images in centos ci image
- unify image building pipelines

### CODE REVIEW

**Review by lentzi90 — COMMENTED** [review]
/approve

File: jenkins/image_building/dib_elements/centos-ci/post-install.d/99-prepull-images [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+# Prepull container images that are later used in CI tests
+set -uex
+
+export CONTAINER_RUNTIME=${CONTAINER_RUNTIME:-"podman"}
+# Container images that we pre-pull into the CI image.
+export IMG_REGISTRY="${IMG_REGISTRY:-"docker.io/registry:latest"}"
+export IMG_GOLANG_IMG="${IMG_GOLANG_IMG:-"docker.io/golang:1.22"}"
```
Comment (kashifest) [inline jenkins/image_building/dib_elements/centos-ci/post-install.d/99-prepull-images:9]: 1.22 ? have we bumped go to 1.22 already?

Comment (Sunnatillo) [reply jenkins/image_building/dib_elements/centos-ci/post-install.d/99-prepull-images]: No, we are not yet using 1.22. 
Changed to 1.21

**Review by kashifest — COMMENTED** [review]
/lgtm

**Pull request merged**